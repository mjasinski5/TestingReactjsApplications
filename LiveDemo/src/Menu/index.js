import React, { Component } from 'react';

class Menu extends Component {
  render() {
    const { isLoggedIn, userName, logout } = this.props;

    return (
      <ul className="menu">
        <li> First Category</li>
        <li> Second Category </li>
        { isLoggedIn &&
          <ul className="additionalElement">ThirdÂ ({userName})
            <button className="logoutButton" onClick={logout}> Logout </button>
          </ul>
        }
      </ul>
    );
  }
}

export default Menu;
