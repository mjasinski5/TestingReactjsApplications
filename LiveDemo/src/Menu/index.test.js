import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Menu from './index';

it('should display menu', () => {

  const menu = shallow(
    <Menu />
  );

  expect(menu.find('.menu').length).toEqual(1);
});

it('should display additional category', () => {
  const isLoggedIn = true;
  const userName = "Mariusz";
  const logoutMock = jest.fn();

  const menu = shallow(
    <Menu isLoggedIn={isLoggedIn} userName={userName} logout={logoutMock}/>
  )

  expect(menu.find('.additionalElement').length).toEqual(1);
  expect(menu.find('.additionalElement').text()).toContain(userName);

  menu.find('.logoutButton').simulate('click');

  expect(logoutMock).toHaveBeenCalled();

});


it('better life with snapshots -  1 :)', () => {
  const isLoggedIn = true;
  const userName = "Mariusz";
  const logoutMock = jest.fn();

  const menu = renderer.create(
    <Menu isLoggedIn={isLoggedIn} userName={userName} logout={logoutMock}/>
  ).toJSON();

  expect(menu).toMatchSnapshot();


});

it('better life with snapshots -  2 :)', () => {
  const isLoggedIn = false;
  
  const menu = renderer.create(
    <Menu isLoggedIn={isLoggedIn} />
  ).toJSON();

  expect(menu).toMatchSnapshot();
});
