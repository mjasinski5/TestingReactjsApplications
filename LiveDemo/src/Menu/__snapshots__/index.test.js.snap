exports[`test better life with snapshots -  1 :) 1`] = `
<ul
  className="menu">
  <li>
     First Category
  </li>
  <li>
     Second Category 
  </li>
  <ul
    className="additionalElement">
    ThirdÂ (
    Mariusz
    )
    <button
      className="logoutButton"
      onClick={[Function]}>
       Logout 
    </button>
  </ul>
</ul>
`;

exports[`test better life with snapshots -  2 :) 1`] = `
<ul
  className="menu">
  <li>
     First Category
  </li>
  <li>
     Second Category 
  </li>
</ul>
`;
